### ISSUE TITLE
[FR] Provide static libraries as well

### ISSUE THREAD

**Issue by bk138** [google/prefab#110]
**Is your feature request related to a problem? Please describe.**
While linking an Android app's native code to .so files works reasonably well, there are a lot of unused symbols shipped in the resulting .apk, contained in the .so files.

**Describe the solution you'd like**
The AAR packages provide .a files as well, allowing linking to those and, as a result, allowing the compiling/linking machinery to strip symbols that aren't actually used by the program code, resulting in smaller .apk's. 

**Describe alternatives you've considered**
Right now we're building all dependencies as static libs from source.

**Additional context**
https://github.com/bk138/multivnc/blob/master/android/app/src/main/cpp/CMakeLists.txt is the project in question - would be nice to use static prefabs there :-).

**Comment by DanAlbert** [comment]
I've been meaning to find the time for this. Is there a particular package you want this for, or just in general?

**Comment by bk138** [comment]
OpenSSL would be nice for a start. Nonetheless, I reckon this would come in handy generally, too.

**Comment by alexcohn** [comment]
Yes, you can use static libraries with prefab. You can have static and shared libraries in the same prefab. The names of the static and shared library files can be the same (e.g. `libssl.so` and `libssl.a`), but the directory names under `modules` and library names in **module.json** must be different (e.g. `libssl` and `libssl-static`)

The only inconvenience is that you must include the headers twice, but @DanAlbert has been convincing me all along that duplication of some files in the AAR is not a problem at all.

**Comment by mrousavy** [comment]
Hi! 

I'd be interested in seeing this feature as well, as I tried to include the OpenSSL library (to include `<openssl/sha.h>`) following [this guide](https://gist.github.com/rhulha/3a2bce7dc3d43c4bab8a14a31feecc38), but can't seem to get it working. While my app builds, it crashes at runtime with this error:

```
    java.lang.UnsatisfiedLinkError: dlopen failed: library "libcrypto.so" not found: needed by /data/app/~~2-M8lYDV9CwRUTUVJ-xI7Q==/com.MYAPPNKq4Bj7P2ot9_6vUGCPvhQ==/lib/x86/libMYLIB.so in namespace classloader-namespace
```

Any guesses on why this happens?

**Comment by bk138** [comment]
> Hi!
> 
> I'd be interested in seeing this feature as well, as I tried to include the OpenSSL library (to include `<openssl/sha.h>`) following [this guide](https://gist.github.com/rhulha/3a2bce7dc3d43c4bab8a14a31feecc38), but can't seem to get it working. While my app builds, it crashes at runtime with this error:
> 
> ```
> java.lang.UnsatisfiedLinkError: dlopen failed: library "libcrypto.so" not found: needed by /data/app/~~2-M8lYDV9CwRUTUVJ-xI7Q==/com.MYAPPNKq4Bj7P2ot9_6vUGCPvhQ==/lib/x86/libMYLIB.so in namespace classloader-namespace
> ```
> 
> Any guesses on why this happens?

Recognizing this is getting off-topic, this is how I did it: https://github.com/bk138/multivnc/commit/9a024da51230b84a76b416de6260a4a27850bf5f
